docker_test.go

```diff
@@ -660,12 +806,118 @@ func TestWebhook(t *testing.T) {
 		},
 		"timestamp": "2020-02-02T00:23:09.544Z"
 	  }`)
-	SendEvent(payload_2) //sending duplicate event to check dedup
+	SendEvent(payload_2, "identify", writeKey) //sending duplicate event to check dedup
+
+	// Sending Batch event
+	payload_Batch := strings.NewReader(`{
+		"batch":
+		[
+			{
+				"userId": "identified_user_id",
+				"anonymousId": "anonymousId_1",
+				"type": "identify",
+				"context":
+				{
+					"traits":
+					{
+						"trait1": "new-val"
+					},
+					"ip": "14.5.67.21",
+					"library":
+					{
+						"name": "http"
+					}
+				},
+				"timestamp": "2020-02-02T00:23:09.544Z"
+			}
+		]
+	}`)
+	SendEvent(payload_Batch, "batch", writeKey)
+
+	// Sending track event
+	payload_track := strings.NewReader(`{
+		"userId": "identified_user_id",
+		"anonymousId":"anonymousId_1",
+		"messageId":"messageId_1",
+		"type": "track",
+		"event": "Product Reviewed",
+		"properties": {
+		  "review_id": "12345",
+		  "product_id" : "123",
+		  "rating" : 3.0,
+		  "review_body" : "Average product, expected much more."
+		}
+	  }`)
+	SendEvent(payload_track, "track",writeKey)
+
+	// Sending page event
+	payload_page := strings.NewReader(`{
+		"userId": "identified_user_id",
+		"anonymousId":"anonymousId_1",
+		"messageId":"messageId_1",
+		"type": "page",
+		"name": "Home",
+		"properties": {
+		  "title": "Home | RudderStack",
+		  "url": "http://www.rudderstack.com"
+		}
+	  }`)
+	SendEvent(payload_page, "page",writeKey)
+
+	// Sending screen event
+	payload_screen := strings.NewReader(`{
+		"userId": "identified_user_id",
+		"anonymousId":"anonymousId_1",
+		"messageId":"messageId_1",
+		"type": "screen",
+		"name": "Main",
+		"properties": {
+		  "prop_key": "prop_value"
+		}
+	  }`)
+	SendEvent(payload_screen, "screen",writeKey)
+
+	// Sending alias event
+	payload_alias := strings.NewReader(`{
+		"userId": "identified_user_id",
+		"anonymousId":"anonymousId_1",
+		"messageId":"messageId_1",
+		"type": "alias",
+		"previousId": "name@surname.com",
+		"userId": "12345"
+	  }`)
+	SendEvent(payload_alias, "alias",writeKey)
+
+	// Sending group event
+	payload_group := strings.NewReader(`{
+		"userId": "identified_user_id",
+		"anonymousId":"anonymousId_1",
+		"messageId":"messageId_1",
+		"type": "group",
+		"groupId": "12345",
+		"traits": {
+		  "name": "MyGroup",
+		  "industry": "IT",
+		  "employees": 450,
+		  "plan": "basic"
+		}
+	  }`)
+	SendEvent(payload_group, "group",writeKey)
+	SendPixelEvents(writeKey)
+
 	require.Eventually(t, func() bool {
-		return len(webhook.Requests()) == 2
+		return len(webhook.Requests()) == 10
+	}, time.Minute, 30*time.Millisecond)
+
+	i := -1
+	require.Eventually(t, func() bool {
+		i = i + 1
+		req := webhook.Requests()[i]
+		body, _ := io.ReadAll(req.Body)
+		return gjson.GetBytes(body, "anonymousId").Str == "anonymousId_1"
```

Comment:
I am not sure of this logic here, doesn't seem safe to me

Reply:
Tested the approach, works well.
But yes ideally it's not the optimal code.
Will think of some other implementation and include in upcoming PR.